Scripts/RemediationScripts/Remediate-SetClassicFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,996 @@
+        Write-Host "`[Step 4 of 4] Set minimum required TLS Version for FrontDoors"
+        Write-Host $([Constants]::SingleDashLine)
+        # To hold results from the remediation.
+        $customDomainsRemediated = @()
+        $frontDoorEPsSkipped = @()
+        
+        $nonCompliantEndpoints | ForEach-Object {
+            $frontdoorEndpoint = $_
+            $resourceId = $_.ResourceID
+            $resourceGroupName = $_.ResourceGroupName
+            $resourceName = $_.ResourceName
+            $hostName = $_.HostName
+            $minimumTlsVersion = $_.MinimumTlsVersion
+            $isMinTlsVersionSetOnCustomDomain = $_.isMinTLSVersionSetOnCustomDomain
+            $isMinTLSVersionSetOnCustomDomainPostRemediation = $_.MinimumTlsVersion;
+            
+            $frontdoorEndpoint | Add-Member -NotePropertyName isMinTLSVersionSetOnCustomDomainPostRemediation -NotePropertyValue $isMinTLSVersionSetOnCustomDomainPostRemediation
+            $frontdoorEndpoint | Add-Member -NotePropertyName PreviousMinimumTlsVersion -NotePropertyValue $MinimumTlsVersion
+
+            Write-Host "Setting minimum required TLS Version for  Front Door : Resource ID: [$($resourceId)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            
+            if (-not [System.Convert]::ToBoolean($isMinTlsVersionSetOnCustomDomain))
+            {
+                try
+                {
+                    Write-Host "Setting minimum required TLS version on custom domain :" $hostName -ForegroundColor $([Constants]::MessageType.Info)
+                    Write-Host $([Constants]::SingleDashLine)
+                    
+                    $resource = Enable-AzFrontDoorCustomDomainHttps -ResourceGroupName $resourceGroupName -FrontDoorName $resourceName -FrontendEndpointName $hostName -MinimumTlsVersion $requiredMinTLSVersion
+                    
+                    if($resource.MinimumTlsVersion -eq $requiredMinTLSVersion)
+                    {
+                        $frontdoorEndpoint.isMinTLSVersionSetOnCustomDomainPostRemediation = $true
+                        $frontdoorEndpoint.minimumTlsVersion = $resource.MinimumTlsVersion
+                        $customDomainsRemediated += $frontdoorEndpoint
+                        Write-Host "Minimum required TLS version for Front Door endpoint has been set successfully for :" $hostName -ForegroundColor $([Constants]::MessageType.Update)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                    else
+                    {
+                        $frontDoorsSkipped += $frontdoorEndpoint
+                        $logResource = @{}
+                        $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                        $logResource.Add("ResourceName",($_.ResourceName))
```

Comment:
Are we logging front door resource or endpoint here?

Reply:
It has been renamed to frontDoorsEpsSkipped